pkg/binlog/position.go

```diff
@@ -346,3 +338,19 @@ func compareIndex(lhs, rhs int) int {
 func (l *Location) ResetSuffix() {
 	l.Suffix = 0
 }
+
+// SetGTID set new gtid for location
+func (l *Location) SetGTID(gset mysql.GTIDSet) error {
```

Comment:
How about adding a UT for read/write with a shallow copy? (to simulate passing a location to `job` or `handle-error`)

Reply:
add ut in f1b2f52. Pointers need to be treated with care.